nixos/modules/installer/tools/nixos-prepare-root.sh

```diff
@@ -30,25 +30,39 @@ if ! test -e "$mountPoint"; then
     exit 1
 fi
 
+setupDir() {
+    mode="$1"
+    shift
+
+    for dir in $@; do
+	if [ -d "$dir" ]; then
+	    chmod "$mode" "$dir"
+	else
+	    mkdir -m "$mode" -p "$dir"
+	fi
+    done
+}
+
+
 # Create a few of the standard directories in the target root directory.
-mkdir -m 0755 -p $mountPoint/dev $mountPoint/proc $mountPoint/sys $mountPoint/etc $mountPoint/run $mountPoint/home
-mkdir -m 01777 -p $mountPoint/tmp
-mkdir -m 0755 -p $mountPoint/tmp/root
-mkdir -m 0755 -p $mountPoint/var
-mkdir -m 0700 -p $mountPoint/root
```

Comment:
I think the `install` command does both creating a directory and change permission in one shot.

Reply:
You are right, `install -d` does that.